### PR TITLE
Add /skills command

### PR DESCRIPTION
**Pull request by umago**
## Summary
This change introduce a /skills slash command to Goose.

While it's possible to ask Goose conversationally for a list of skills, the results are not consistent because they depend on the underlying model. For someone using different models for different purposes or environments (work and personal) this list of skill can be presented in varying formats and structures each time.

With the /skills command it's always reliable and consistent. Here's the usage:

/skills                   - list available skills with name and description
/skills [`skill`, ...]  - load one or more skills into the session
                         (e.g. /skills code-review insight)

Tab-completion for skill names is supported (e.g. /skills co`[tab]` -> code-review)

### Testing
* Unit tests available

### Related Issues
Relates to #8599

### Screenshots/Demos (for UX changes)
Before:  N/A

After:   
```
🪿 /skills
+---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------+
| Skill               | Description                                                                                                                                         |
+===========================================================================================================================================================================+
| code-review         | Senior engineer code review focused on catching issues before they become PR comments. Reviews only changed lines, categorizes issues by priority,  |
|                     | and fixes them one by one. Use when the user says "code review", "review my code", "review this branch", or wants pre-PR feedback.                  |
|---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------|
| coding              | Strict coding guidelines biasing toward caution, precision, and minimalism. Use for all coding tasks to ensure clean, surgical changes.             |
|---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------|
| create-app-e2e-test | Create app e2e tests for the Goose desktop app using the Tauri app test driver. Use when the user wants to generate, write, or verify UI tests that |
|                     | run against the live app.                                                                                                                           |
|---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------|
| create-pr           | Create a GitHub PR from the current branch: handle uncommitted changes, generate a summary, and submit via gh CLI. Use when the user says "create   |
|                     | PR", "open PR", "submit PR", "push PR", or wants to create a pull request.                                                                          |
|---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------|
| edge-case-finder    | Analyzes branch changes to find edge cases, error states, and untested user flow paths in UI code. Use when the user says "find edge cases", "what  |
|                     | am I missing", "edge cases", "test my flows", "what could go wrong", or wants to harden a feature before shipping.                                  |
|---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------|
| goose-doc-guide     | Reference goose documentation to create, configure, or explain goose-specific features like recipes, extensions, sessions, and providers. You MUST  |
|                     | fetch relevant goose docs before answering. You MUST NOT rely on training data or assumptions for any goose-specific fields, values, names, syntax, |
|                     | or commands.                                                                                                                                        |
|---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------|
| insight             | Direct, casual, and focused on mental models and analogies.                                                                                         |
|---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------|
| repo-scan           | Guidelines for analyzing an unfamiliar codebase. Prioritizes high-level architecture, structural clarity, and providing entry points for            |
|                     | deep-dives.                                                                                                                                         |
|---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------|
| sys-design          | Architect-level skill for generating a Technical Design Doc (TDD) intended as the primary input for coding agents and human review.                 |
+---------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------+
  ╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌ 0% 0/1.0M
🪿 /skills coding insight

  ────────────────────────────────────────
  ▸ load_skill
    name: coding

  ────────────────────────────────────────
  ▸ load_skill
    name: insight

Both skills loaded. Here's what's active:

**coding** — Strict guidelines for minimal, precise, surgical changes. Zero scope creep, test-driven verification, laser focus on what's requested.

**insight** — Direct, casual tone. Teach the *why*, not just the *what*. Use analogies and mental models. No fluff.

Ready to work.

```

### CODE REVIEW

**Comment by michaelneale** [comment]
thanks @umago - do you want to look in ui/text for the next gen TUI there as well - as it could be the place for this

**Comment by umago** [comment]
Hi @michaelneale, sounds good. I'm still getting familiar with the codebase, but I’ll dig into it.

File: crates/goose-cli/src/session/input.rs [hunk 1]
```diff
@@ -267,6 +270,16 @@ fn handle_slash_command(input: &str) -> Option<InputResult> {
...
         s if s.starts_with(CMD_RECIPE) => parse_recipe_command(s),
         s if s == CMD_COMPACT => Some(InputResult::Compact),
+        // Match "/skills" exactly or "/skills " with args - avoids matching e.g. "/skillsextra"
+        s if s == CMD_SKILLS || s.starts_with(&format!("{CMD_SKILLS} ")) => {
+            let args = s.get(CMD_SKILLS.len()..).unwrap_or("").trim();
+            if args.is_empty() {
+                Some(InputResult::ListSkills)
+            } else {
+                let names: Vec<String> = args.split_whitespace().map(String::from).collect();
+                Some(InputResult::LoadSkills(names))
```
[the remark this answers is not in the document]
Comment (umago) [reply crates/goose-cli/src/session/input.rs:280]: Skills should not have spaces like that, see: https://platform.claude.com/docs/en/agents-and-tools/agent-skills/overview

The name field "Must contain only lowercase letters, numbers, and hyphens"

File: crates/goose-cli/src/session/mod.rs [hunk 2]
```diff
@@ -874,6 +882,55 @@ impl CliSession {
...
+        result?;
+
+        Ok(())
+    }
+
+    async fn handle_list_skills(&mut self) -> Result<()> {
+        use comfy_table::{presets, Cell, ContentArrangement, Table};
+        use goose::agents::platform_extensions::skills::list_installed_skills;
+        let cwd = std::env::current_dir().unwrap_or_default();
+        let skills = list_installed_skills(Some(&cwd));
```
[the remark this answers is not in the document]
Comment (umago) [reply crates/goose-cli/src/session/mod.rs:911]: Yes. Skills can exist both locally and globally, the ones in the current directory should only be listed when you are working inside that specific project.

**Comment by umago** [comment]
@michaelneale I was just reading about the new TUI in the ui/text/README.md. Looks promising, I will dig more into it and play with it a bit (probably after work). Thanks for the tip, happy to collaborate with it.

**Review by lifeizhou-ap — APPROVED** [review]
Nice one! Thanks for your contribution

File: crates/goose-cli/src/session/mod.rs [hunk 3]
```diff
@@ -874,6 +882,55 @@ impl CliSession {
...
+                .collect::<Vec<_>>()
+                .join(", ")
+        );
+        self.push_message(Message::user().with_text(&message));
+        output::show_thinking();
+        let result = self
+            .process_agent_response(true, CancellationToken::default())
+            .await;
+        output::hide_thinking();
+        result?;
```
[the remark this answers is not in the document]
Comment (umago) [reply crates/goose-cli/src/session/mod.rs:902]: That's the idea, delegate the load of the skill to the specific tool. That way it's consistent, whether you do it conversationally or via /skills, it uses the same tool and you get consistent feedback about it being loaded or not from the tool already.

**Comment by umago** [comment]
@lifeizhou-ap thank you for the review!
